RFC/src/RFC-0130_Mining.md

```diff
@@ -1 +1,92 @@
+
+The solution to the PoW problem could be solved at the difficulty of either the Tari and/or Monero blockchain. If a solution has been found that meets the minimum difficulty requirements of the Monero blockchain, then the new Tari block should be updated by including the solved Monero block’s information (block header hash, merkel tree branch, and hash of the coinbase transaction) into the PoW summary section of the Tari block header. If a solution was found that meet the Tari or Monero blockchain difficulty then the new Tari block can be added to the Tari blockchain. If the solution met the difficulty requirements of the Monero blockchain then the new Monero block can also be added to the Monero blockchain. If the PoW solution was sufficient to meet the difficult level of both the Tari and Monero blockchains then the individual blocks for each cryptocurrency can be added to their respective blockchains.
+
+Solved and completed blocks should be propagated to the rest of the base layer network so that Nakamoto consensus can be performed on the new block and it can be added to the local blockchain copies of the full mining nodes.
+
+####  Primary functionality required by a Tari full-node miner
+- The Tari blockchain should have the ability to be merged mined with Monero using the CryptoNight Proof-of-Work algorithm. 
+- The Tari full-node miner must maintain complete or pruned copies of the Tari and Monero blockchains.
+- It must be able to transmit and propagate information on the Tari base layer network using peer-to-peer communication using a gossip protocol. It could also propagate information and blocks on the Monero network.
+- It should have a mechanism to construct a new Tari and Monero block by selecting transactions from the different Tari and Monero mempools that need to be included in the different blocks.
+- It must have the ability to include the block header hash of the new Tari block into the coinbase section of a newly created Monero block to enable merged mining.
+- It must be able to include the Monero block header hash, merkel tree branch, and hash of the coinbase transaction of the Monero block into the PoW summary field of the new Tari block header. 
+- It must have the ability to perform a PoW algorithm on the newly created Monero block, that contains the Tari block information. It should then attempting to solve the PoW algorithm at the desired Tari or Monero difficulty.  
+- It must have the functionality to verify the validity of a newly received or created block for either the Tari and/or Monero blockchain. Created or received valid Tari blocks should be included in the local Tari blockchain. Created or received valid Monero blocks should be included in the local Monero blockchain.
+- Valid Tari and/or Monero blocks should be propagated to other mining nodes on the Tari base layer network and/or the Monero network. 
```

Comment:
Again, this is what the base nodes will do. The mining code only does mining.

Reply:
This RFC is probably more appropriate as an RFC for a "base layer node" or it might have to be split up a bit, see my suggestion below.. What I called a full-node miner is actually a base layer node. The requirements were derived from the tasks of a node provided in the original bitcoin paper on page 3 (https://bitcoin.org/bitcoin.pdf). Also, I called it a full-node miner (for a lack of a better name) to try and differentiate it from a mining worker that is only receiving PoW tasks from the mining pool operator.  In this example, the mining pool operator is actually a full-node miner or base layer node that is only distributing his PoW tasks.

I think it might be appropriate to break down the base-layer into 3 parts:
- Base Layer Node: Maintains a Tari blockchain and distributes completed Blocks and valid Transactions on Base layer network.
- Mining Node: Constructs blocks from transactions in mempool of connected Base Layer Node, can perform local PoW or can distribute PoW to Mining Workers, and distribute solved blocks to connected Base layer Node.
- Mining Worker: Does not construct blocks but only perform PoW tasks for a connected Mining Node.